livekit-agents/livekit/agents/stt/stt.py

```diff
@@ -39,13 +39,23 @@ class SpeechEvent:
     alternatives: List[SpeechData] = field(default_factory=list)
 
 
+@dataclass
+class STTCapabilities:
+    streaming_supported: bool
+    interim_results: bool
+
+
 class STT(ABC):
     def __init__(
         self,
         *,
-        streaming_supported: bool,
+        capabilities: STTCapabilities,
```

Comment:
should this be required? that'll break all existing plugins?

Reply:
i think we should give at least one version for deprecating this and still keeping the API open. maybe throw a warn